tests/test_download.py

```diff
@@ -0,0 +1,80 @@
+from luminoso_api.v5_client import LuminosoClient
+from luminoso_api.v5_download import iterate_docs, download_docs
+
+import pytest
+import requests
+import tempfile
+import json
+
+BASE_URL = 'http://mock-api.localhost/api/v5/'
+RESPONSE = {
+    'result': [
+        {'title': 'Document 1', 'text': 'hello', 'tokens': [{'term_id': 'hello|en'}], 'metadata': [], 'vector': 'AAAA', 'doc_id': 'uuid-0'},
+        {'title': 'Document 2', 'text': 'hello', 'tokens': [{'term_id': 'hello|en'}], 'metadata': [], 'vector': 'AAAA', 'doc_id': 'uuid-1'},
+    ],
+    'total_count': 2
+}
+
+EXPANDED_DOCS = [
+    {'title': 'Document 1', 'text': 'hello', 'tokens': [{'term_id': 'hello|en'}], 'metadata': [], 'vector': 'AAAA'},
+    {'title': 'Document 2', 'text': 'hello', 'tokens': [{'term_id': 'hello|en'}], 'metadata': [], 'vector': 'AAAA'},
+]
+
+CONCISE_DOCS = [
+    {'title': 'Document 1', 'text': 'hello', 'metadata': []},
+    {'title': 'Document 2', 'text': 'hello', 'metadata': []},
+]
+
+REPETITIVE_DOC = {'title': 'Yadda', 'text': 'yadda yadda', 'metadata': []}
+
+
+def test_iteration(requests_mock):
+    """
+    Test iterating over the documents in a project.
+    """
+    requests_mock.get(BASE_URL + 'projects/projid/docs/', json=RESPONSE)
+    client = LuminosoClient.connect(BASE_URL + 'projects/projid', token='fake')
+
+    docs = list(iterate_docs(client, progress=False))
+    assert docs == CONCISE_DOCS
+
+    docs = list(iterate_docs(client, progress=False, expanded=True))
+    assert docs == EXPANDED_DOCS
+
+
+def test_pagination(requests_mock):
+    """
+    Test iterating over 1002 documents that come in two pages.
+    """
+    client = LuminosoClient.connect(BASE_URL + 'projects/projid', token='fake')
+    page1 = [REPETITIVE_DOC for i in range(1000)]
+    page2 = [REPETITIVE_DOC for i in range(2)]
```

Comment:
What's the reason for doing it this way, instead of `[REPETITIVE_DOC] * 1000` like you did below?  I think the only difference is that in the one case you get 1000 references to the same dictionary, and in the other case you get 1000 copies of the same dictionary.  Since we're not comparing using `is` it should work fine either way.